agreementbot/secrets/plugin_errors.go

```diff
@@ -0,0 +1,174 @@
+package secrets
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/open-horizon/anax/cli/cliutils"
+)
+
+// ----- ERROR WRAPPER -----
+
+// overall VaultError structure that wraps the below errors; every error coming out of the vault (which will be one
```

Comment:
Can we change VaultError to SecretsProviderError? I would like to get rid of references to vault unless specifically interacting with the hashicorp vault. The plugin sh
ould be agnostic.